gulpfile.js

```diff
@@ -151,20 +151,28 @@ gulp.task('default', function() {
 // download all the installer dependencies so we can package them up into the .exe
 gulp.task('prefetch', function(cb) {
   let counter=0;
+  var reqObjs = [];
   for (var key in reqs) {
     if (reqs.hasOwnProperty(key)) {
       let currentUrl = reqs[key].url;
       let currentKey = key;
+
       // download only what can be included in offline installer
       if (reqs[key].bundle === 'yes') {
         counter++;
-          console.log('DOWNLOADING -> ' + reqs[key].url);
-          request(reqs[key].url)
-            .pipe(fs.createWriteStream(path.join(prefetchFolder, key))).on('finish',function() {
-            counter--;
-            if(counter===0) {
-              cb();
-            }
+        reqObjs[currentKey] = request(reqs[currentKey].url, function (err) {
+          if (err) { console.log('DOWNLOADING -> ' + reqs[currentKey].url); console.log(err); console.log(""); }
+        });
+        reqObjs[currentKey].pipe(fs.createWriteStream(path.join(prefetchFolder, currentKey))).on('finish',function() {
+          counter--;
+          if(counter===0) {
+            cb();
+          }
+        });
+        reqObjs[currentKey].on('response', function(res){

```

Comment:
Started out fine but then...

$➔ gulp prefetch
[11:34:11] Using gulpfile ~\Documents\GitHub\developer-platform-install\gulpfile.js
[11:34:11] Starting 'prefetch'...
DOWNLOADING 57 Kb -> http://cdk-builds.usersys.redhat.com/builds/09-Mar-2016/cdk-2.0.0-beta5.zip
DOWNLOADING 981 Mb -> http://cdk-builds.usersys.redhat.com/builds/09-Mar-2016/rhel-cdk-kubernetes-7.2-21.x86_64.vagrant-virtualbox.box
DOWNLOADING 351 Kb -> http://the.earth.li/~sgtatham/putty/latest/x86/pscp.exe
DOWNLOADING 73 Mb -> http://cdn.azul.com/zulu/bin/zulu8.13.0.5-jdk8.0.72-win_x64.zip
DOWNLOADING 857 Kb -> https://cygwin.com/setup-x86_64.exe
DOWNLOADING 12 Mb -> https://github.com/openshift/origin/releases/download/v1.1.4/openshift-origin-client-tools-v1.1.4-3941102-windows.zip
DOWNLOADING 518 Mb -> https://devstudio.redhat.com/9.0/snapshots/builds/devstudio.product_9.0.mars/latest/all/jboss-devstudio-9.1.0.latest-installer-standalone.jar
buffer.js:389
    throw new Error('toString failed');
    ^

Error: toString failed
    at Buffer.toString (buffer.js:389:11)
    at BufferList.toString (C:\Users\nboldt\Documents\GitHub\developer-platform-install\node_modules\request\node_modules\bl\bl.js:166:33)
    at Request.<anonymous> (C:\Users\nboldt\Documents\GitHub\developer-platform-install\node_modules\request\request.js:1014:32)
    at emitOne (events.js:82:20)
    at Request.emit (events.js:169:7)
    at IncomingMessage.<anonymous> (C:\Users\nboldt\Documents\GitHub\developer-platform-install\node_modules\request\request.js:963:12)
    at emitNone (events.js:72:20)
    at IncomingMessage.emit (events.js:166:7)
    at endReadableNT (_stream_readable.js:905:12)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickCallback (node.js:356:17)

*sad trombone*